packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx

```diff
@@ -9,16 +9,14 @@ import TabItem from '@theme/TabItem';
 >
 > See **https://typescript-eslint.io/rules/no-wrapper-object-types** for documentation.
 
-The JavaScript language includes a set of primitives such as `boolean` and `number`s.
-JavaScript also contains a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
-TypeScript represents the primitive in lower-case types and instances of the wrapper classes with upper-case.
+The JavaScript language has a set of language types, but some of them correspond to two TypeScript types, which look similar: `boolean`/`Boolean`, `number`/`Number`, `string`/`String`, `bigint`/`BigInt`, `symbol`/`Symbol`, `object`/`Object`.
+The difference is that the lowercase variants are compiler intrinsics and specify the actual _runtime types_ (that is, the return value when you use the `typeof` operator), while the uppercase variants are _structural types_ defined in the library that can be satisfied by any user-defined object with the right properties, not just the real primitives.
+JavaScript also has a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
+These wrapper objects are assignable to the uppercase types, but not to the lowercase types.
 
-Using the primitives like `0` instead of class constructors like `Number(0)` is generally considered a JavaScript best practice.
-As a result, using the lower-case primitive type names like `number` instead of class names like `Number` is generally considered a TypeScript best practice.
-
-Similarly, TypeScript's `Function` type refers to objects created with the `Function` constructor, i.e. `new Function("eval'd code with implicit args")`.
-`Function` allows being called with any number of arguments and returns type `any`.
-It's generally better to specify function parameter and return types with function type syntax.
+Using the primitives like `0` instead of object wrappers like `new Number(0)` is generally considered a JavaScript best practice.
+JavaScript programs typically work with the real number primitives, rather than objects that "look like" numbers.
```

Comment:
might find a way to mention that this is _required_ for equality to work correctly, (as well as truthiness/falsiness but that's less important)